warehouse/query-core/src/main/java/datawave/query/tables/ShardQueryLogic.java

```diff
@@ -577,6 +580,11 @@ protected String getStopwatchHeader(ShardQueryConfiguration config) {
     
     @Override
     public QueryLogicTransformer getTransformer(Query settings) {
+        if (this.transformerInstance != null) {
+            addConfigBasedTransformers();
+            return this.transformerInstance;
+        }
```

Comment:
The creation code was always in the `getTransformer()` method -- I just added the add the null condition. Also in keeping with the singleton patter, the getInstnace() usually does the creation if the instance is null, but it could be delegated as you suggest.  Kind of a half a dozen here half a dozen there style thing.